src/Renci.SshNet.Tests/Classes/Connection/SocketFactoryTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Net;
+using System.Net.Sockets;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Renci.SshNet.Connection;
+
+namespace Renci.SshNet.Tests.Classes.Connection
+{
+    [TestClass]
+    public class SocketFactoryTests
+    {
+        [TestMethod]
+        public void CreateOnSocketFactoryWithLocalIpAddressShouldHaveLocalEndpoint()
+        {
+            var socketFactory = new SocketFactory();
+
+            var socket = socketFactory.Create(
+                AddressFamily.InterNetwork,
+                SocketType.Stream,
+                ProtocolType.Tcp,
+                IPAddress.Loopback);
```

Comment:
Other Tests are failing after this one runs because you're not Disposing the socket.
You can perhaps look at the DirectConnectorTest_Connect_* testcases and add this one to that collection, inheriting from DirectConnectorTestBase and overriding Teardown() function.

Reply:
Hmm, actually, I think tests are failing due to port exhaustion on the Appveyor VM - sockets remain in TIME_WAIT state for a long time. This is happening randomly on other PRs that have nothing to do with sockets.

To avoid this the SocketFactory code may need a flag to tell it to use SO_REUSEADDR on the sockets. Or if possible, reducing the TIME_WAIT delay or increasing the pool of available dynamic ports on the VM. 

This is outside of your PR.

Reply:
so whom should be the person to reconfigure the VM as you mentioned this is outside of the PR

Reply:
@drieseng has access to it, not sure if anyone else has.